### PR TITLE
Add the uninstall hook to the docs

### PR DESCRIPTION
**Pull request by kagarmoe**
Signed-off-by: kagarmoe <kgarmoe@chef.io>

Replaces the uninstall content.

### CODE REVIEW

File: components/docs-chef-io/content/habitat/application_lifecycle_hooks.md [hunk 1]
```diff
@@ -142,3 +142,13 @@ File location: `<plan>/hooks/post-run`. The post run hook will get executed afte
 ### post-stop
 
 File location: `<plan>/hooks/post-stop`. The post-stop hook will get executed after service has been stopped successfully. You may use this hook to undo what the `init` hook has done.
+
+n `install` hook, unlike other hooks with the exception of the `uninstall` hook, will not have access to any census data exposed via binds or the `svc` namespace. Also, configuration in `svc_config_path` is not accessible to an `install` hook. If an `install` hook needs to use templated configuration files, templates located in the `svc_config_install_path` may be referenced. This location will contain rendered templates in a package's `config_install` folder. Finally, any configuration updates made during a service's runtime that would alter an `install` hook or any configuration template in `svc_config_install_path` will not cause a service to reload.
```
Comment (IanMadd) [inline components/docs-chef-io/content/habitat/application_lifecycle_hooks.md:146]:
```suggestion
An `install` hook, unlike other hooks with the exception of the `uninstall` hook, will not have access to any census data exposed via binds or the `svc` namespace. Also, configuration in `svc_config_path` is not accessible to an `install` hook. If an `install` hook needs to use templated configuration files, templates located in the `svc_config_install_path` may be referenced. This location will contain rendered templates in a package's `config_install` folder. Finally, any configuration updates made during a service's runtime that would alter an `install` hook or any configuration template in `svc_config_install_path` will not cause a service to reload.
```

**Review by IanMadd — APPROVED** [review]
One typo, otherwise it looks good.
tenor-137173448

File: components/docs-chef-io/content/habitat/application_lifecycle_hooks.md (hunk 1 above)
Comment (mwrock) [reply components/docs-chef-io/content/habitat/application_lifecycle_hooks.md:146]: Was this paragraph intended to be duplicated from further up in the doc. It seems out of place here.

File: components/docs-chef-io/content/habitat/application_lifecycle_hooks.md [hunk 2]
```diff
@@ -142,3 +142,13 @@ File location: `<plan>/hooks/post-run`. The post run hook will get executed afte
...
+
+#### uninstall
```
Comment (mwrock) [inline components/docs-chef-io/content/habitat/application_lifecycle_hooks.md:148]: Should be `###` instead of `####`

**Review by mwrock — CHANGES_REQUESTED** [review]

**Review by mwrock — APPROVED** [review]

**Pull request merged**